docs/sphinx/api/public_distributed_api.rst

```diff
@@ -177,3 +177,64 @@ Member functions
    +----------------------------------------------------------+
    | :cpp:func:`hpx::distributed::latch::wait`                |
    +----------------------------------------------------------+
+
+``hpx/components.hpp``
+======================
+
+The header :hpx-header:`libs/full/include/include,hpx/include/components.hpp`
+includes the components implementation. A component in `hpx` is a C++ class
+which can be created remotely and for which its member functions can be invoked
+remotely as well. More information about how components can be defined,
+created, and used can be found in :ref:`components`.
+
+Macros
+------
+
+.. table:: `hpx` macros of header ``hpx/components.hpp``
+
+   +----------------------------------------------+
+   | Macro                                        |
+   +==============================================+
+   | :c:macro:`HPX_REGISTER_ACTION`               |
+   +----------------------------------------------+
+   | :c:macro:`HPX_REGISTER_COMPONENT`            |
+   +----------------------------------------------+
+   | :c:macro:`HPX_REGISTER_COMMANDLINE_REGISTRY` |
+   +----------------------------------------------+
+   | :c:macro:`HPX_REGISTER_COMMANDLINE_OPTIONS`  |
+   +----------------------------------------------+
+   | :c:macro:`HPX_DEFINE_COMPONENT_ACTION`       |
+   +----------------------------------------------+
+   | :c:macro:`HPX_REGISTER_ACTION_DECLARATION`   |
+   +----------------------------------------------+
```

Comment:
I think we should document the following macros:

For component modules we need:

`HPX_REGISTER_COMPONENT_MODULE`
`HPX_REGISTER_COMPONENT`

For component actions we need:

`HPX_DEFINE_COMPONENT_ACTION`

as well as:

`HPX_REGISTER_ACTION_DECLARATION`
`HPX_REGISTER_ACTION`

For the command line option support, we need:

```
// Register a function to be called to populate the special command line          
// options supported by this component.                                           
HPX_REGISTER_COMMANDLINE_MODULE
```

Additionally, there is also:
```
// Register a startup function which will be called as a HPX-thread during
// runtime startup.
HPX_REGISTER_STARTUP_MODULE
```

I don't think `HPX_REGISTER_COMMANDLINE_REGISTRY` has to be documented.